waspc/data/Generator/templates/react-app/src/router.tsx

```diff
@@ -25,35 +27,31 @@ export const routeNameToRouteComponent = {
   {=/ routes =}
 } as const;
 
-const router = (
-  <Router basename="{= baseDir =}">
-    {=# rootComponent.isDefined =}
-    <{= rootComponent.importIdentifier =}>
-    {=/ rootComponent.isDefined =}
-    <Switch>
-      {=# isExternalAuthEnabled =}
-      {/* 
-        Wasp specific routes *must* go first to prevent user
-        defined routes from overriding them.
-        Details in https://github.com/wasp-lang/wasp/issues/2029
-      */}
-      <Route exact path="{= oAuthCallbackPath =}">
-        <OAuthCallbackPage />
-      </Route>
-      {=/ isExternalAuthEnabled =}
-      {Object.entries(routes).map(([routeKey, route]) => (
-        <Route
-          exact
-          key={routeKey}
-          path={route.to}
-          component={routeNameToRouteComponent[routeKey]}
-        />
-      ))}
-    </Switch>
-    {=# rootComponent.isDefined =}
-    </{= rootComponent.importIdentifier =}>
-    {=/ rootComponent.isDefined =}
-  </Router>
-)
-
-export default router
+const waspDefinedRoutes = [
+  {=# isExternalAuthEnabled =}
+  {
+    path: "{= oAuthCallbackPath =}",
+    Component: OAuthCallbackPage,
+  },
+  {=/ isExternalAuthEnabled =}
+]
+const userDefinedRoutes = Object.entries(routes).map(([routeKey, route]) => {
+  return {
+    path: route.to,
+    Component: routeNameToRouteComponent[routeKey],
+  }
+})
+
+const browserRouter = createBrowserRouter([{
+  path: '/',
+  {=# rootComponent.isDefined =}
+  element: <{= rootComponent.importIdentifier =} />,
```

Comment:
Pleas make sure the root component still works as expected. I'm guessing you did, but mentioning just in case.

Reply:
Yes, I have tested having the root component, not having it at all. It works as expected in all the example apps.